lib/ContractInteract/PricerRule.js

```diff
@@ -0,0 +1,348 @@
+      return Promise.reject(err);
+    }
+    if (!Web3.utils.isAddress(txOptions.from)) {
+      const err = new TypeError(`Invalid from address: ${txOptions.from}.`);
+      return Promise.reject(err);
+    }
+
+    const tx = PricerRule.deployRawTx(
+      auxiliaryWeb3,
+      organization,
+      eip20Token,
+      tokenRules,
+      baseCurrencyCode,
+      conversionRate,
+      conversionRateDecimals,
+      requiredPriceOracleDecimals
+    );
+
+    return Utils.sendTransaction(tx, txOptions).then((txReceipt) => {
+      const address = txReceipt.contractAddress;
+      return new PricerRule(auxiliaryWeb3, address);
+    });
+  }
+
+  /**
+   * Method which returns Tx object to deploy GnosisSafe master copy contract.
+   *
+   * @param {Web3} auxiliaryWeb3 Auxiliary chain web3 object.
+   * @param {String} organization Organization address.
+   * @param {String} eip20Token The economy token address.
+   * @param {String} tokenRules The economy token rules address.
+   * @param {String} baseCurrencyCode The economy base currency code.
+   * @param {String} conversionRate The conversion rate from the economy base currency
+   *                        to the token.
+   * @param {String} conversionRateDecimals The conversion rate's decimals from the
+   *                                economy base currency to the token.
+   * @param {String} requiredPriceOracleDecimals Required decimals for price oracles.
+   *
+   * @returns {Object} Raw transaction.
+   */
+  static deployRawTx(
+    auxiliaryWeb3,
+    organization,
+    eip20Token,
+    tokenRules,
+    baseCurrencyCode,
+    conversionRate,
+    conversionRateDecimals,
+    requiredPriceOracleDecimals
+  ) {
+    if (!(auxiliaryWeb3 instanceof Web3)) {
+      throw new TypeError(`Mandatory Parameter 'auxiliaryWeb3' is missing or invalid: ${auxiliaryWeb3}`);
+    }
+    const abiBinProvider = new AbiBinProvider();
+    const bin = abiBinProvider.getBIN(ContractName);
+
+    const bytesBaseCurrencyCode = this.auxiliaryWeb3.utils.stringToHex(baseCurrencyCode.toString());
+    const args = [
+      organization,
+      eip20Token,
+      bytesBaseCurrencyCode,
+      conversionRate,
+      conversionRateDecimals,
+      requiredPriceOracleDecimals,
+      tokenRules
+    ];
+    const contract = Contracts.getPricerRule(auxiliaryWeb3);
+
+    return contract.deploy({
+      data: bin,
+      arguments: args
+    });
+  }
+
+  /**
+   * Adds a new price oracle. From address should be only worker address.
+   *
+   * @param {String} priceOracleAddress PriceOracle contract address.
+   * @param {Object} txOptions Tx options.
+   *
+   * @return {Promise<Object>} Promise that resolves to transaction receipt.
+   */
+  async addPriceOracle(priceOracleAddress, txOptions) {
+    if (!txOptions) {
+      const err = new TypeError(`Invalid transaction options: ${txOptions}.`);
+      return Promise.reject(err);
+    }
+    if (!Web3.utils.isAddress(txOptions.from)) {
```

Comment:
I think it's optional. If we don't provide gasPrice EVM takes a default value.

Reply:
I think we need some discussion here. The validation should also be applied in mosaic.js, bt.js also. Instead of replicating this validation in every method we can refactor to a common place. Will update in the ticket.